addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactHandler.java

```diff
@@ -786,29 +783,36 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 }
                 case CHANNEL_ENABLED: {
                     if (command instanceof OnOffType) {
-
                         if (command == OnOffType.ON) {
                             sendCommand("ena 1");
                         } else if (command == OnOffType.OFF) {
                             sendCommand("ena 0");
                         } else {
                             return;
                         }
-
                     }
                     break;
                 }
                 case CHANNEL_OUTPUT: {
                     if (command instanceof OnOffType) {
-
                         if (command == OnOffType.ON) {
                             sendCommand("output 1");
                         } else if (command == OnOffType.OFF) {
                             sendCommand("output 0");
                         } else {
                             return;
                         }
-
+                    }
+                    break;
+                }
+                case CHANNEL_DISPLAY: {
+                    if (command instanceof StringType) {
+                        if (type == KebaType.P30 && (series == KebaSeries.C || series == KebaSeries.X)) {
+                            int maxLength = (command.toString().length() < 23) ? command.toString().length() : 23;
+                            sendCommand("display 0 0 0 0 " + command.toString().substring(0, maxLength));
+                        } else {
+                            logger.warn("'Display' is not supported on a Keba {}:{}", type, series);
```

Comment:
Display is not supported on all Keba's, so any invalid usage should be logged. I preferred not to create 10 different thing types but rather differentiate between version and firmware versions in a single Handler